libraries/wasm-jit/Source/Runtime/LLVMJIT.cpp

```diff
@@ -215,69 +218,48 @@ namespace LLVMJIT
 	{
 		JITUnit(bool inShouldLogMetrics = true)
 		: shouldLogMetrics(inShouldLogMetrics)
-		#ifdef _WIN32
-			, pdataCopy(nullptr)
-		#endif
 		{
-			objectLayer = llvm::make_unique<ObjectLayer>(NotifyLoadedFunctor(this),NotifyFinalizedFunctor(this));
+			objectLayer = llvm::make_unique<llvm::orc::LegacyRTDyldObjectLinkingLayer>(ES,[](llvm::orc::VModuleKey K) {
+                           return llvm::orc::LegacyRTDyldObjectLinkingLayer::Resources{
+                              std::make_shared<UnitMemoryManager>(), std::make_shared<llvm::orc::NullResolver>()
+                              };
+                       },
+                       [](llvm::orc::VModuleKey, const llvm::object::ObjectFile &Obj, const llvm::RuntimeDyld::LoadedObjectInfo &o) {
+                          //nothing to do
+                       },
+                       [this](llvm::orc::VModuleKey, const llvm::object::ObjectFile &Obj, const llvm::RuntimeDyld::LoadedObjectInfo &o) {
+                           for(auto symbolSizePair : llvm::object::computeSymbolSizes(Obj)) {
+                              auto symbol = symbolSizePair.first;
+                              auto name = symbol.getName();
+                              auto address = symbol.getAddress();
+                              if(symbol.getType() && symbol.getType().get() == llvm::object::SymbolRef::ST_Function && name && address) {
+                                 Uptr loadedAddress = Uptr(*address);
+                                 auto symbolSection = symbol.getSection();
+                                 if(symbolSection)
+                                    loadedAddress += (Uptr)o.getSectionLoadAddress(*symbolSection.get());
+                                 //printf(">>> %s 0x%lX\n", symbolSizePair.first.getName()->data(), loadedAddress);
+                                 std::map<U32,U32> offsetToOpIndexMap;
```

Comment:
This parameter looks dead to me.  I couldn't find any other place that calls notifySymbolLoaded, and nothing ever sets the map to non-empty.  Am I missing something?

Reply:
You are correct. Originally, WAVM's `notifySymbolLoaded()` would consume this and create debug information mapping each wasm opcode to the x86 instruction(s) it generated. As you can imagine that created quite the memory bloat 😳 I removed that code from `notifySymbolLoaded()` but decided not to refactor away the function signature as I was trying to minimize the changes to WAVM. Since WAVM will be deprecated in 2.0 I would rather not continue hacking it up further here.